lms/static/scripts/frontend_apps/components/FilePickerFormFields.js

```diff
@@ -45,6 +45,7 @@ export default function FilePickerFormFields({
       {Object.entries(formFields).map(([field, value]) => (
         <input key={field} type="hidden" name={field} value={value} />
       ))}
+      {groupSet && <input type="hidden" name="group_set" value={groupSet} />}
```

Comment:
I still want to confirm that this hidden field is necessary. I'm kind of taking it on faith right now.

Reply:
It's necessary as things stand now.

- The input field is necessary to send "us" back the groupSet of the assignment and then we can store it on the DB. This is the case in blackboard and likely any other non-Canvas LMS we implemented this in.

- While on canvas we send this information to canvas itself, but we encode it on  that `contentItem` structure via `extraParams`. When later the assigment is launched the `groups_set` query param would be present in the launch URL.

It's true that we are sending the same information in two places no matter in what LMS we are at. We can't control how canvas behaves but we could in the backend instead of using this new hidden field parse the `contentItem` field

```
{"@context":"http://purl.imsglobal.org/ctx/lti/v1/ContentItem","@graph":[{"@type":"LtiLinkItem","mediaType":"application/vnd.ims.lti.v1.ltilink","url":"http://localhost:8001/lti_launches?group_set=_64_1&url=https%3A%2F%2Fexample.com"}]}
```

and extract the `group_set` query param by parsing URL. I personally think that's clunkier than a bit of duplication but maybe it's something we could look into.

Pinging @seanh  for comments here.

Reply:
Yeah :man_shrugging: Canvas works one way because we use LTI content item selection in Canvas, whereas every other LMS works a different way because we don't use content item selection. I think let's just leave it for now